akka-actor-tests/src/test/scala/akka/pattern/BackoffSupervisorSpec.scala

```diff
@@ -47,27 +64,92 @@ class BackoffSupervisorSpec extends AkkaSpec with ImplicitSender {
     }
 
     "forward messages to the child" in {
-      val supervisor = system.actorOf(
-        BackoffSupervisor.props(Child.props(testActor), "c2", 100.millis, 3.seconds, 0.2))
-      supervisor ! "hello"
-      expectMsg("hello")
+      def assertForward(supervisor: ActorRef) = {
+        supervisor ! "hello"
+        expectMsg("hello")
+      }
+      assertForward(create(onStopOptions()))
+      assertForward(create(onFailureOptions()))
     }
 
-    "support custom supervision decider" in {
-      val supervisor = system.actorOf(
-        BackoffSupervisor.propsWithSupervisorStrategy(Child.props(testActor), "c1", 100.millis, 3.seconds, 0.2,
-          OneForOneStrategy() {
-            case _: TestException ⇒ SupervisorStrategy.Stop
-          }))
-      supervisor ! BackoffSupervisor.GetCurrentChild
-      val c1 = expectMsgType[BackoffSupervisor.CurrentChild].ref.get
-      watch(c1)
-      c1 ! "boom"
-      expectTerminated(c1)
-      awaitAssert {
+    "support custom supervision strategy" in {
+      def assertCustomStrategy(supervisor: ActorRef) = {
         supervisor ! BackoffSupervisor.GetCurrentChild
-        // new instance
-        expectMsgType[BackoffSupervisor.CurrentChild].ref.get should !==(c1)
+        val c1 = expectMsgType[BackoffSupervisor.CurrentChild].ref.get
+        watch(c1)
+        c1 ! "boom"
+        expectTerminated(c1)
+        awaitAssert {
+          supervisor ! BackoffSupervisor.GetCurrentChild
+          // new instance
+          expectMsgType[BackoffSupervisor.CurrentChild].ref.get should !==(c1)
+        }
+      }
+      filterException[TestException] {
+        val stoppingStrategy = OneForOneStrategy() {
+          case _: TestException ⇒ SupervisorStrategy.Stop
+        }
+        val restartingStrategy = OneForOneStrategy() {
+          case _: TestException ⇒ SupervisorStrategy.Restart
+        }
+
+        assertCustomStrategy(create(onStopOptions().withSupervisorStrategy(stoppingStrategy)))
+        assertCustomStrategy(create(onFailureOptions().withSupervisorStrategy(restartingStrategy)))
+      }
+    }
+
+    "support default stopping strategy when using `BackoffOptions.onStop`" in {
+      filterException[TestException] {
+        val supervisor = create(onStopOptions().withDefaultStoppingStrategy)
+        supervisor ! BackoffSupervisor.GetCurrentChild
+        val c1 = expectMsgType[BackoffSupervisor.CurrentChild].ref.get
+        watch(c1)
+        c1 ! "boom"
+        expectTerminated(c1)
+        awaitAssert {
+          supervisor ! BackoffSupervisor.GetCurrentChild
+          // new instance
+          expectMsgType[BackoffSupervisor.CurrentChild].ref.get should !==(c1)
```

Comment:
maybe also check that the restarts count was incremented by sending `GetRestartCount`?